File: courses/codes/cclv1-jr-class/in-class-cc-1-sol/leap.cpp
#include <iostream>
using namespace std;

int main() {
    int year;
    bool isLeap = false;
    cin >> year;
    
    if (year % 400 == 0) {
        isLeap = true;
    } else if (year % 100 == 0) {
        isLeap = false;
    } else if (year % 4 == 0) {
        isLeap = true;
    }

    if (isLeap) {
        cout << year << " is a leap year." << endl;
    } else {
        cout << year << " is not a leap year." << endl;
    }

    return 0;
}

File: courses/codes/cclv1-jr-class/in-class-cc-1-sol/leap.py
year = int(input())
is_leap = False

if year % 400 == 0:
    is_leap = True
elif year % 100 == 0:
    is_leap = False
elif year % 4 == 0:
    is_leap = True
    
if is_leap:
    print(str(year) + " is a leap year.")
else:
    print(str(year) + " is not a leap year.")